exchanges/kraken.py

```diff
@@ -25,15 +30,20 @@ def order_info(self, order_id):
     def trades_history(self, asset_pair, start_time, finish_time):
         pass
 
-    def _public_query(self, query_name, parameters=None):
+    def _public_query(self, query_name, data=None):
         """Queries public data on Kraken"""
 
         # Request public data. This can raise an exception
         # if there is a connection error or if the request times
         # out. In addition, raise an exception if the status is not
         # OK.
         url = '{}public/{}'.format(self.url, query_name)
-        response = requests.get(url, timeout=self.timeout)
+        response = requests.post(url, data=data, timeout=self.timeout)
```

Comment:
I agree, there's no downside, but the "get" method expresses better what we want to do: to get public data. And I'm not talking only about the action of "getting something", I'm talking about the action of simply asking for something public, with no server side-effect, and which doesn't need any security measure.

Respect to the Invalid parameters error.. requests.get(url, params=data) should work, I can try if you want.